gateway-service-impl/src/main/java/org/hypertrace/gateway/service/entity/query/DataFetcherNode.java

```diff
@@ -19,13 +19,26 @@ public class DataFetcherNode implements QueryNode {
   private Integer limit;
   private Integer offset;
   private List<OrderByExpression> orderByExpressionList = Collections.emptyList();
-
   private final boolean canFetchTotal;
 
+  private final QueryNode childNode;
+
   public DataFetcherNode(String source, Filter filter) {
     this.source = source;
     this.filter = filter;
-    this.canFetchTotal = false; // total would be computed in memory
+    this.childNode = new NoOpNode();
+
+    // total would be computed in memory
+    this.canFetchTotal = false;
+  }
+
+  public DataFetcherNode(String source, Filter filter, QueryNode childNode) {
+    this.source = source;
+    this.filter = filter;
+    this.childNode = childNode;
```

Comment:
Is there any restriction or limitation on the type of data a childNode fetch node can have?